Library/Formula/agda.rb

```diff
@@ -21,24 +22,70 @@ class Agda < Formula
   head "https://github.com/agda/agda.git", :branch => "master"
 
   option "without-epic-backend", "Exclude the 'epic' compiler backend"
+  option "without-stdlib", "Don't install the Agda standard library"
 
   depends_on "cabal-install" => :build
   depends_on "ghc"
   depends_on "gmp"
   depends_on "bdw-gc" if build.with? "epic-backend"
 
+  if build.with? "stdlib"
+    if build.head?
+      resource "stdlib" do
+        url "https://github.com/agda/agda-stdlib.git", :branch => "master"
+      end
+    else
+      resource "stdlib" do
+        url "https://github.com/agda/agda-stdlib/archive/v0.9.tar.gz"
+        sha1 "f39d5685ab2dc47758c87d9068047fce6b4b99a1"
+      end
+    end
+  end
+
   def install
+    # set the appropriate flag for using or not using the Epic backend
     if build.with? "epic-backend"
       epic_flag = "-fepic"
     else
       epic_flag = "-f-epic"
     end
+
+    # install Agda itself
     cabal_sandbox do
       cabal_install_tools "alex", "happy", "cpphs"
+      if build.with? "epic-backend"
+        cabal_install "--prefix=#{prefix}" "epic"
+      end
       cabal_install "--only-dependencies", epic_flag
       cabal_install "--prefix=#{prefix}", epic_flag
     end
     cabal_clean_lib
+
+    # install the Agda standard library, if we're supposed to
+    if build.with? "stdlib"
+      resource("stdlib").stage prefix/"agda-stdlib"
+      cd prefix/"agda-stdlib" do # install the standard library's helper tools
+        cabal_sandbox do
+          cabal_install "--only-dependencies"
+          cabal_install "--prefix=#{prefix/"agda-stdlib"}"
+          system prefix/"agda-stdlib"/"bin"/"GenerateEverything"
+        end
+        cabal_clean_lib
+      end
+
+      # this next bit doesn't play well with the relocatablility of bottles,
+      # so it gets excluded if we're building one
+      unless build.bottle?
+        cd prefix/"agda-stdlib"/"ffi" do # install the standard library's FFI bindings
+          cabal_install "--user"
+        end
```

Comment:
This should be excluded by default rather than just in a bottle or it's confusing for users that the bottle will be installed with a weird missing thing. Let's just make it an option instead.